lib/Bio/EnsEMBL/DataCheck/Checks/DisplaybleGenes.pm

```diff
@@ -0,0 +1,69 @@
+=head1 LICENSE
+
+Copyright [2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::DisplaybleGenes;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Utils qw/sql_count/;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'DisplaybleGenes',
+  DESCRIPTION    => 'Check that genes are displayable and '
+    . 'have web_data attached to their analysis.',
+  GROUPS         => ['core_handover'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['core', 'otherfeatures', 'rnaseq', 'cdna'],
+  TABLES         => ['gene', 'analysis', 'analysis_description']
+};
+
+sub tests {
+  my ($self) = @_;
+
+  sub skip_tests {
+    my ($self) = @_;
+
+    my $sql = 'SELECT COUNT(*) FROM analysis_description';
+
+    if (! sql_count($self->dba, $sql) ) {
+      return (1, 'No analysis_description');
+    }
+  }
+
+  my $sql = q/
+    SELECT COUNT(gene_id)
```

Comment:
Using count is fine, but I just wanted to point out that you get bonus diagnostics if you do a non-count select. For example, if the test fails, you probably want to know the analysis it failed for. If you did `SELECT DISTINCT logic_name`, the offending logic_names will be reported in the test failure.
The disadvantage is that this query will take a bit longer than the count, but usually it's a matter of milliseconds, so that's not normally an issue.
If you do it this was, it's best to add a diagnostic message; by default it'll say 

> Unexpected data: (logic_name) = (ensembl_genomes)

but you can replace that first bit with something more helpful, like "Undisplayed analysis", by setting that as an additional `$diag` parameter, and passing it as the last option to `is_rows_zero`.
(See https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/SequenceLevel.pm for an example)